doc/reference/peripherals/reset.rst

```diff
@@ -0,0 +1,23 @@
+.. _reset_api:
+
+Reset Controller
+################
+
+Overview
+********
+
+Reset controllers are units that control the reset signals to multiple peripherals. The reset controller API allows peripheral drivers to request control over their reset input signals, including the ability to assert, deassert and toggle those signals. Also, the reset status of the reset input signal can be checked.
+
+Mainly, the assert and deassert API functions are optional because in most cases we want to toggle the reset signals.
```

Comment:
nit: please wrap to 80cols as most docs do

Reply:
add the API in https://docs.zephyrproject.org/latest/reference/api/overview.html

Reply:
@gmarull Experimental?

Reply:
Yep, sounds like it should be experimental since there'll be just 1 implementation at the beginning, so the API may need adjustments as more are added.

Reply:
Okay, cool.